tiledb/sm/c_api/tiledb.h

```diff
@@ -42,6 +42,15 @@
 #include <stdint.h>
 #include <stdio.h>
 
+/* ****************************** */
+/*            CONSTANTS           */
+/* ****************************** */
+
+#define TILEDB_FILTER_GZIP_DEFAULT_LEVEL -1
```

Comment:
We should have 1 defined value I think which is `COMPRESSOR_DEFAULT_LEVEL` or similar. I don't think we need one per compressor. This simplifies the user interface/experience for setting default value. @nguyenv what do you think?

Reply:
I think the easiest way is to overload the compress function so that it doesn't need to make in a compressor level at all and just uses the default value. In that case, we might not need any of these constants defined in `tiledb.h`.

Reply:
@nguyenv so we should leave the calls to `default_level()` with the default values and hard-code those same values into the overloaded function, scrapping the defined constants? I think the original idea behind the constants was to allay confusion for users who tend to use `-1` as default values now that the `ZStd::default_level()` has changed to allow negative values. But I agree that the overloaded function should be all that's necessary - it'll use the default without potentially adding the confusion of the new defined constants. What do you think @Shelnutt2